### PR TITLE
Feature/many block files

### PR DESCRIPTION
**Pull request by cshcomcom**
## Background
- Related with `AIM-017: Directory structure for many block files`
- Linux file management tools have limit on the number of files in a directory
- So we need a directory structure for block files 

## Description
- Now, n2b and h2n have prefix directory. (`/h2n/0x???/{block_hash}`, `/n2b/{block_number / 100000}/{block_number}`)

Resolve: #468

### CODE REVIEW

File: common/constants.js [hunk 1]
```diff
@@ -71,6 +71,8 @@ if (!fs.existsSync(BLOCKCHAIN_DATA_DIR)) {
 const CHAINS_DIR = path.resolve(BLOCKCHAIN_DATA_DIR, 'chains');
 const CHAINS_N2B_DIR_NAME = 'n2b'; // NOTE: Block number to block.
 const CHAINS_H2N_DIR_NAME = 'h2n'; // NOTE: Block hash to block number.
+const CHAINS_N2B_NUMBER_OF_MAX_FILES = 100000;
```
Comment (liayoo) [inline common/constants.js:74]: minor) how about CHAINS_N2B_MAX_NUMBER_OF_FILES or CHAINS_N2B_NUMBER_OF_FILES_MAX ?

**Review by liayoo — APPROVED** [review]
LGTM with a minor comment. Thanks!

File: common/constants.js (hunk 1 above)
Comment (platfowner) [reply common/constants.js:74]: or CHAINS_N2B_MAX_NUM_FILES?

File: common/file-util.js [hunk 2]
```diff
@@ -120,6 +132,10 @@ class FileUtil {
   static writeBlock(chainPath, block) {
     const blockPath = this.getBlockPath(chainPath, block.number);
     if (!fs.existsSync(blockPath)) {
+      const blockDirPath = this.getBlockDirPath(chainPath, block.number);
+      if (block.number % CHAINS_N2B_NUMBER_OF_MAX_FILES === 0 && !fs.existsSync(blockDirPath)) {
```
Comment (platfowner) [inline common/file-util.js:136]: What about just `if (!fs.existsSync(blockDirPath)) {`?

File: common/file-util.js [hunk 3]
```diff
@@ -120,6 +132,10 @@ class FileUtil {
...
+        fs.mkdirSync(this.getBlockDirPath(chainPath, block.number));
```
Comment (platfowner) [inline common/file-util.js:137]: `fs.mkdirSync(blockDirPath);`?

File: common/constants.js (hunk 1 above)
Comment (cshcomcom) [reply common/constants.js]: I changed it to CHAINS_N2B_MAX_NUM_FILES! Thanks

File: common/file-util.js (hunk 2 above)
Comment (cshcomcom) [reply common/file-util.js]: Done!

File: common/file-util.js (hunk 3 above)
Comment (cshcomcom) [reply common/file-util.js]: I missed it! Thanks

**Comment by cshcomcom** [comment]
All comments are addressed! PTAL @liayoo @platfowner

**Review by platfowner — APPROVED** [review]
LGTM
Thanks!

**Comment by cshcomcom** [comment]
Thansk for the reviews! Now merging

**Pull request merged**